packages/squiggle-lang/src/rescript/Utility/XYShape.res

```diff
@@ -542,10 +541,10 @@ module Range = {
     (nextY -. lastY) /. (nextX -. lastX)
 
   let mapYsBasedOnRanges = (fn, t) =>
-    Belt.Array.zip(t.xs, t.ys)
+    E.A.zip(t.xs, t.ys)
     |> E.A.toRanges
     |> E.R.toOption
-    |> E.O.fmap(r => r |> Belt.Array.map(_, r => (nextX(r), fn(r))))
+    |> E.O.fmap(r => r |> E.A.fmap(_, r => (nextX(r), fn(r))))
```

Comment:
`r->E.A.fmap(r => ...)` would be cleaner. (but you probably haven't done that to avoid mixing arrow styles in the same code?)

Reply:
I ended up with `->E.O.fmap(r => r->E.A.fmap(r => (nextX(r), fn(r))))` once I went through XYShape removing `|>`s, so, agreed!